src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/AspireEFPostgreSqlExtensions.cs

```diff
@@ -113,34 +206,31 @@ public static partial class AspireEFPostgreSqlExtensions
                     NpgsqlCommon.AddNpgsqlMetrics(meterProviderBuilder);
                 });
         }
+    }
 
-        void ConfigureDbContext(DbContextOptionsBuilder dbContextOptionsBuilder)
+    private static void ConfigureDbContext(NpgsqlEntityFrameworkCorePostgreSQLSettings settings, DbContextOptionsBuilder dbContextOptionsBuilder, Action<DbContextOptionsBuilder>? configureDbContextOptions)
+    {
+        if (!settings.Retry)
         {
-            // delay validating the ConnectionString until the DbContext is requested. This ensures an exception doesn't happen until a Logger is established.
-            if (string.IsNullOrEmpty(settings.ConnectionString))
-            {
-                throw new InvalidOperationException($"ConnectionString is missing. It should be provided in 'ConnectionStrings:{connectionName}' or under the 'ConnectionString' key in '{DefaultConfigSectionName}' or '{typeSpecificSectionName}' configuration section.");
-            }
-
-            // We don't register a logger factory, because there is no need to: https://learn.microsoft.com/dotnet/api/microsoft.entityframeworkcore.dbcontextoptionsbuilder.useloggerfactory?view=efcore-7.0#remarks
-            dbContextOptionsBuilder.UseNpgsql(settings.ConnectionString, builder =>
-            {
-                // Resiliency:
-                // 1. Connection resiliency automatically retries failed database commands: https://www.npgsql.org/efcore/misc/other.html#execution-strategy
-                if (settings.MaxRetryCount > 0)
-                {
-                    builder.EnableRetryOnFailure(settings.MaxRetryCount);
-                }
-                // 2. "Scale proportionally: You want to ensure that you don't scale out a resource to a point where it will exhaust other associated resources."
-                // The pooling is enabled by default, the min pool size is 0 by default: https://www.npgsql.org/doc/connection-string-parameters.html#pooling
-                // There is nothing for us to set here.
-                // 3. "Timeout: Places limit on the duration for which a caller can wait for a response."
-                // The timeouts have default values, except of Internal Command Timeout, which we should ignore:
-                // https://www.npgsql.org/doc/connection-string-parameters.html#timeouts-and-keepalive
-                // There is nothing for us to set here.
-            });
-
-            configureDbContextOptions?.Invoke(dbContextOptionsBuilder);
+            return;
         }
+
+        // We don't provide the connection string, it's going to use the pre-registered DataSource.
+        // We don't register logger factory, because there is no need to: https://learn.microsoft.com/dotnet/api/microsoft.entityframeworkcore.dbcontextoptionsbuilder.useloggerfactory?view=efcore-7.0#remarks
+        dbContextOptionsBuilder.UseNpgsql(builder =>
```

Comment:
I think you always need to call `UseNpgsql` whether settings.Retry is true or false.

Reply:
Right, mistake when I refactored the code to reuse the same method for both Enrich and AddNpgsql. And with your changes I might have to split them.

Reply:
It also means we might be missing a test that should have caught that for AddNpgsql.

Reply:
Done